core/src/main/java/nl/nn/adapterframework/pipes/IncreaseIntegerPipe.java

```diff
@@ -39,20 +43,37 @@
 
 	private String sessionKey=null;
 	private int increment=1;
+	private final static String PARAMETER_INCREMENT = "increment";
 
 	@Override
 	public void configure() throws ConfigurationException {
 		super.configure();
-		if (StringUtils.isEmpty(sessionKey))
+		if (StringUtils.isEmpty(sessionKey)) {
 			throw new ConfigurationException("sessionKey must be filled");
+		}
 	}
 
 	@Override
 	public PipeRunResult doPipe(Message message, IPipeLineSession session) throws PipeRunException {
 
 		String sessionKeyString = (String) session.get(sessionKey);
 		Integer sessionKeyInteger = Integer.valueOf(sessionKeyString);
-		session.put(sessionKey, sessionKeyInteger.intValue() + increment + "");
+
+		int incrementBy = increment;
+		ParameterValueList pvl = null;
+		if (getParameterList() != null) {
+			try {
+				pvl = getParameterList().getValues(message, session);
+			} catch (ParameterException e) {
+				throw new PipeRunException(this, getLogPrefix(session) + "exception extracting parameters", e);
+			}
+		}
+		ParameterValue pv = pvl.getParameterValue(PARAMETER_INCREMENT);
+		if(pv != null) {
+			incrementBy = pv.asIntegerValue(increment);
+		}
```

Comment:
The first line will fail with NPE when parameterList == null. 
Either remove the test `` getParameterList() != null ``, or put these lines inside the guarded block